File: conteudo-curso/aulas-visual-basic/Pratica 0912/Hospital3.0_blank/Hospital_3/BancoDeDados.vb
' Importação da Biblioteca do Firebird
Imports FirebirdSql.Data.FirebirdClient

Public Class BancoDeDados
    ' Atributos
    Private conexao As New FbConnection
    Private comando As New FbCommand
    Private caminho As String = Application.StartupPath & "\DB_HOSPITAL.GDB"
    Private transacao As FbTransaction

    ' Método Conectar
    Public Sub Conectar()
        Dim str_conexao As String = "User=SYSDBA;Password=masterkey;Database=" & caminho & ";DataSource=localhost;Port=3050;"
        me.conexao.ConnectionString = str_conexao
        Me.conexao.Open()
    End Sub

    ' Método Executar
    Public Sub Executar(sql As String, ParamArray parametros() As Object)
        Me.comando.Connection = Me.conexao
        Me.comando.CommandText = sql
        Me.comando.Parameters.Clear()
        For Each parametro In parametros
            Me.comando.Parameters.Add(New FbParameter With {.Value = parametro})
        Next
        Me.comando.ExecuteNonQuery()
    End Sub

    ' Método Consultar
    Public Function Consultar(sql As String, ParamArray parametros() As Object) As DataTable
        Me.comando.Connection = Me.conexao
        Me.comando.CommandText = sql
        Me.comando.Parameters.Clear()
        For Each parametro In parametros
            Me.comando.Parameters.Add(New FbParameter With {.Value = parametro})
        Next
        Dim dados As FbDataReader = Me.comando.ExecuteReader()
        Dim tabela As New DataTable
        tabela.Load(dados)
        dados.Close()
        Return tabela
    End Function

    ' Método Desconectar
    Public Sub Desconectar()
        Me.conexao.Close()
    End Sub
End Class

File: conteudo-curso/aulas-desenv-web/Aula 1410/sistema_av_diagnostico_pratica/src/class/BancoDeDados.php
<?php
    class BancoDeDados {
        // Atributos
        private $conexao;
        private $dbName = 'av_diag_pratica';
        private $dbUser = 'root';
        private $dbPassword = 'masterkey';

        // Método Construtor
        public function __construct() {
            $this->conexao = new PDO('mysql:host=localhost;port=3307;dbname=' . $this->dbName, $this->dbUser, $this->dbPassword);
        }

        // Método Executar Comando
        public function executarComando($sql, $parametros = []) {
            $stmt = $this->conexao->prepare($sql);
            $stmt->execute($parametros);
        }

        // Método Consultar
        public function consultar($sql, $parametros = [], $fetchAll = false) {
            $stmt = $this->conexao->prepare($sql);
            $stmt->execute($parametros);
            
            if ($fetchAll) {
                return $stmt->fetchAll(PDO::FETCH_ASSOC);
            } else {
                return $stmt->fetch(PDO::FETCH_ASSOC);
            }
        }
    }